alica_engine/src/engine/collections/AssignmentCollection.cpp

```diff
@@ -51,81 +87,111 @@ shared_ptr<vector<const supplementary::AgentID*>> AssignmentCollection::getRobot
+}
+
 void AssignmentCollection::sortEps() {
-    vector<EntryPoint*> sortedEpVec;
-    for (short i = 0; i < this->numEps; i++) {
-        sortedEpVec.push_back(this->entryPoints[i]);
-    }
-    stable_sort(sortedEpVec.begin(), sortedEpVec.end(), EntryPoint::compareTo);
-    for (short i = 0; i < this->numEps; i++) {
-        this->entryPoints[i] = sortedEpVec.at(i);
+    std::stable_sort(_entryPoints, _entryPoints + _numEps, EntryPoint::compareTo);
+}
+
+void AssignmentCollection::sortRobots(const EntryPoint* ep) {  // TODO: make obsolete by maintaining sortedness
+    for (int i = 0; i < _numEps; ++i) {
+        if (_entryPoints[i] == ep) {
+            std::sort(_robotIds[i].begin(), _robotIds[i].end(), supplementary::AgentIDComparator());
+        }
     }
 }
 
 /**
  * Removes all robots from the AssignmentCollection
  */
 void AssignmentCollection::clear() {
-    for (int i = 0; i < this->numEps; i++) {
-        this->robotIds[i]->clear();
+    for (int i = 0; i < _numEps; ++i) {
+        _robotIds[i].clear();
     }
 }
 
-string AssignmentCollection::toString() {
-    stringstream ss;
-    for (int i = 0; i < this->numEps; i++) {
-        if (this->entryPoints[i] != nullptr) {
-            ss << this->entryPoints[i]->getId() << " : ";
-            for (auto& robotId : *this->robotIds[i]) {
-                ss << robotId << ", ";
+std::string AssignmentCollection::toString() const {
+    std::stringstream ss;
+    for (int i = 0; i < _numEps; ++i) {
+        if (_entryPoints[i] != nullptr) {
+            ss << _entryPoints[i]->getId() << " : ";
+            for (const supplementary::AgentID* robotId : _robotIds[i]) {
+                ss << *robotId << ", ";
             }
-            ss << endl;
+            ss << std::endl;
         }
     }
     return ss.str();
 }
 
 short AssignmentCollection::getSize() const {
-    return this->numEps;
+    return _numEps;
 }
 
 void AssignmentCollection::setSize(short size) {
-    this->numEps = size;
+    _numEps = size;
 }
 
-EntryPoint* AssignmentCollection::getEp(short index) {
-    if (index < this->numEps) {
-        return this->entryPoints[index];
+const EntryPoint* AssignmentCollection::getEp(short index) const {
+    if (index < _numEps) {
+        return _entryPoints[index];
     } else {
         return nullptr;
     }
 }
 
-bool AssignmentCollection::setEp(short index, EntryPoint* ep) {
-    if (index < this->numEps) {
-        this->entryPoints[index] = ep;
-        return true;
-    } else {
-        cerr << "AssCol: Index to HIGH!!!!!! ########################################" << endl;
-        return false;
-    }
+void AssignmentCollection::setEp(short index, const EntryPoint* ep) {
+    assert(index < _numEps);
+    _entryPoints[index] = ep;
 }
 
+void AssignmentCollection::addRobot(const supplementary::AgentID* id, const EntryPoint* e) {
+    for (int i = 0; i < _numEps; ++i) {
+        if (_entryPoints[i] == e) {
+            _robotIds[i].push_back(id);
```

Comment:
the way addRobot and removeRobot are implemented tells me that _robotIds should be a map instead of a vector.